vortex-array/src/arrays/constant/compute/take.rs

```diff
@@ -0,0 +1,66 @@
+use vortex_error::VortexResult;
+use vortex_mask::AllOr;
+use vortex_scalar::Scalar;
+
+use crate::arrays::{ConstantArray, ConstantEncoding};
+use crate::builders::{ArrayBuilderExt, builder_with_capacity};
+use crate::compute::TakeFn;
+use crate::{Array, ArrayRef};
+
+impl TakeFn<&ConstantArray> for ConstantEncoding {
+    fn take(&self, array: &ConstantArray, indices: &dyn Array) -> VortexResult<ArrayRef> {
+        match indices.validity_mask()?.boolean_buffer() {
+            AllOr::All => {
+                Ok(ConstantArray::new(array.scalar().clone(), indices.len()).into_array())
+            }
+            AllOr::None => Ok(ConstantArray::new(
+                Scalar::null(array.dtype().clone()),
+                indices.len(),
+            )
+            .into_array()),
+            AllOr::Some(v) => {
```

Comment:
If the constant array is already a null value, you can also return constant?

Reply:
Sure, can add a check for it